src/test/setup.js

```diff
@@ -24,6 +24,8 @@ afterEach(() => {
   //
   // See https://github.com/facebook/jest/issues/7654
   jest.resetAllMocks();
+  jest.restoreAllMocks();
+  jest.clearAllTimers();
```

Comment:
Without clearing timers, we can have weird interactions between tests. I was very surprised this wasn't done by default...

Reply:
Huh, that's interesting. But is that causing a problem when we don't clear those timers? Current `setTimeout`s are without any delay and we use `runAllTimers` anyway.

Reply:
I think I wasn't running `runAllTimers` in all of the cases, and as a result the timers were run in the next test running `runAllTimers`...
That's why it's easiest to do it this way :) This way we don't need to `runAllTimers` in all tests when we don't need to, we just need to have `useFakeTimers` at the start.